### PR TITLE
Feature/exclude packages

### PR DESCRIPTION
**Pull request by sdhh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,18 @@ When run from the command line, `owdit` will inspect the dependencies listed in
 When vulnerabilities are found, `owdit` prints out a pretty-formatted report.  
 The exit code of `owdit` is the number of found vulnerabilities or _-1_ on error.
 
+##### Ignoring vulnerabilities in specific packages
+When desirable, one can specify packages to be excluded from owdit's check in a `.owditrc` file in the same folder as `package.json`:
+
+```json
+{
+  "excludes": [ "foo", "bar" ]
```
Comment (KurtPattyn) [inline README.md]: Comma missing?

File: lib/owdit.js [hunk 2]
```diff
@@ -2,9 +2,46 @@ const exec = require("executive");
...
 const util = require("util");
+const fs = require("fs");
+const path = require("path");
+const rcFilename = ".owditrc";
+
+// Read from .owditrc the lists of packages that should be excluded or only generate warnings
+function readWarnsAndExcludes(cwd) {
+  let warns = [];
+  let excludes = [];
+  const rcFile = path.resolve(cwd, rcFilename);
```
Comment (KurtPattyn) [inline lib/owdit.js]: Why not just `require` the .owditrc file?

File: lib/owdit.js [hunk 3]
```diff
@@ -2,9 +2,46 @@ const exec = require("executive");
...
+  if (excludes.length) {
+    console.log(`Vulnerabilities in these packages will be IGNORED: ${excludes.join(", ")}.`)
+  }
+
+  return {warns, excludes};
+}
 
 module.exports.check = function(cwd, callback)
 {
+  const {warns, excludes} = readWarnsAndExcludes(cwd);
```
Comment (KurtPattyn) [inline lib/owdit.js:45]: Here you use `cwd` while the readme and usage says .owditrc should live in the same directory as the package.json file.

File: package.json [hunk 4]
```diff
@@ -1,8 +1,11 @@
 {
   "name": "owdit",
   "description": "Audits dependencies for known vulnerabilities.",
-  "version": "1.0.0",
+  "version": "1.1.0",
   "author": "Pattyn Kurt <pattyn.kurt@gmail.com>",
+  "contributors": [
+    "Long Truong <yypsch@gmail.com>"
+  ],
```
Comment (KurtPattyn) [inline package.json:8]: Of course!

**Review by KurtPattyn — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (sdhh) [reply README.md]: Damned, you even lint JSON in markdown!

File: lib/owdit.js (hunk 2 above)
Comment (sdhh) [reply lib/owdit.js]: Doesn't that mean `.owditrc` needs to be a module (with exports)? I wanted to keep that file as simple as possible, hence plain JSON.

File: lib/owdit.js (hunk 3 above)
Comment (sdhh) [reply lib/owdit.js:45]: I will look into how I can adapt the docs or change the code.

When run as a command, owdit actually does a `npm ls` inside the current working directory, and npm is smart enough to look recursively up in the directory tree for a `package.json` when one is absent from the cwd. So what I documented is erroneous: `.owditrc` will only be taken into consideration when present in the cwd.

File: lib/owdit.js (hunk 2 above)
Comment (sdhh) [reply lib/owdit.js]: So my assumption was wrong: it's technically perfectly possible to `require()` a JSON file, but the file must have a `.json` extension.
I tried to require `.owditrc` and this resulted in an error as node assumed Javascript. `require.extensions` could allow `.owditrc` to be considered as JSON but is now deprecated.
And of course "owdit.json" is not as sexy.

File: lib/owdit.js (hunk 3 above)
Comment (sdhh) [reply lib/owdit.js:45]: Adapted code so owdit looks for a `.owditrc` file in the same dir as `package.json`, even when run in a subdirectory of the project root.

Comment (sdhh) [reply lib/owdit.js:45]: (starts from cwd)

**Review by KurtPattyn — APPROVED** [review]

**Pull request merged**